src/main/java/at/hannibal2/skyhanni/features/inventory/wardrobe/WardrobeAPI.kt

```diff
@@ -0,0 +1,229 @@
+
+    private fun getWardrobeSlotFromId(id: Int?) = slots.find { it.id == id }
+
+    fun inWardrobe() = inventoryPattern.matches(InventoryUtils.openInventoryName())
+
+    fun createPriceLore(slot: WardrobeSlot) = buildList {
+        if (slot.isEmpty()) return@buildList
+        add("§aEstimated Armor Value:")
+        var totalPrice = 0.0
+        for (stack in slot.armor.filterNotNull()) {
+            val price = EstimatedItemValueCalculator.getTotalPrice(stack)
+            add("  §7- ${stack.name}: §6${NumberUtil.format(price)}")
+            totalPrice += price
+        }
+        if (totalPrice != 0.0) add(" §aTotal Value: §6§l${NumberUtil.format(totalPrice)} coins")
+    }
+
+    @SubscribeEvent
+    fun onInventoryUpdate(event: InventoryUpdatedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        inventoryPattern.matchMatcher(event.inventoryName) {
+            currentPage = group("currentPage").formatInt()
+        } ?: return
+        if (currentPage == null) return
+
+        val itemsList = event.inventoryItems
+
+        val allGrayDye = slots.all {
+            itemsList[it.inventorySlot]?.itemDamage == EnumDyeColor.GRAY.dyeDamage || !it.isInCurrentPage()
+        }
+
+        if (allGrayDye) {
+            val allSlotsEmpty = slots.all {
+                (getWardrobeItem(itemsList[it.helmetSlot]) == null && getWardrobeItem(itemsList[it.chestplateSlot]) == null &&
+                    getWardrobeItem(itemsList[it.leggingsSlot]) == null && getWardrobeItem(itemsList[it.bootsSlot]) == null) ||
+                    !it.isInCurrentPage()
+            }
+            if (allSlotsEmpty) {
+                for (slot in slots.filter { it.isInCurrentPage() }) {
+                    slot.getData()?.armor = emptyArmor()
+                }
+            } else return
+        }
+
+        var foundCurrentSlot = false
+
+        for (slot in slots.filter { it.isInCurrentPage() }) {
+            slot.getData()?.armor = listOf(
+                getWardrobeItem(itemsList[slot.helmetSlot]),
+                getWardrobeItem(itemsList[slot.chestplateSlot]),
+                getWardrobeItem(itemsList[slot.leggingsSlot]),
+                getWardrobeItem(itemsList[slot.bootsSlot]),
+            )
+            if (equippedSlotPattern.matches(itemsList[slot.inventorySlot]?.name)) {
+                currentSlot = slot.id
+                foundCurrentSlot = true
+            }
+            slot.locked = (itemsList[slot.inventorySlot] == ItemStack(Items.dye, EnumDyeColor.RED.dyeDamage))
+            if (slot.locked) slots.forEach { if (it.id > slot.id) it.locked = true }
+        }
+        if (!foundCurrentSlot && getWardrobeSlotFromId(currentSlot)?.page == currentPage) {
+            currentSlot = null
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        DelayedRun.runDelayed(500.milliseconds) {
```

Comment:
Check if we were in a custom wardrobe previously, and only then run this delayed task